proposals/exception-handling/Exceptions-formal-overview.md

```diff
@@ -0,0 +1,211 @@
+# 3rd proposal formal spec overview
+
+This is an overview of the 3rd proposal's formal spec additions, to aid in discussions concerning the proposed semantics.
+
+## Abstract Syntax
+
+### Types
+
+Tag Types
+
+```
+tagtype ::= [t*]→[]
+```
+
+### Instructions
+
+```
+instr ::= ... | throw x | rethrow l
+        | try bt instr* (catch x instr*)* (catch_all instr*)? end
+        | try bt instr* delegate l
+```
+
+### Modules
+
+Tags
+
+```
+tag ::= export* tag exntype  | export* tag tagtype import
+```
+
+Modules
+
+```
+mod ::= module ... tag*
+```
+
+## Validation (Typing)
+
+#### Modification to labels
+
+To verify that a `try...delegate l` instruction refers to a label surrounding the instructions of a try block (call this a try-label), introduce a `kind` attribute to labels in the validation context, which is set to `try` when the label is a try-label.
+
+Similarly, to verify that the `rethrow l` instruction refers to a label surrounding the instructions of a catch block (call this a catch-label), we allow the `kind` attribute of labels in the validation context to be set to `catch` when the label is a catch-label.
+
```

Comment:
So, in terms of grammar:
```suggestion
labelkind ::= try | catch
labeltype ::= {result resulttype, kind labelkind?}
C ::= {..., labels labeltype}
```
Maybe extend the labelkind with `block` instead of making it an option.